pkg/controller/clusterrole/clusterrole_controller_test.go

```diff
@@ -0,0 +1,63 @@
+package clusterrole
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/openshift/osd-metrics-exporter/pkg/metrics"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+	"github.com/stretchr/testify/require"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
```

Comment:
I would like to go with `envtest` but it needs a specific version of `kube-apiserver`, `etcd` and other binaries to work. And since this doesn't exist in the build environment yet I thought I will keep these tests for now.